compiler/src/dotty/tools/dotc/transform/ElimRepeated.scala

```diff
@@ -73,18 +74,23 @@ class ElimRepeated extends MiniPhaseTransform with InfoTransformer with Annotati
     transformTypeOfTree(tree)
 
   override def transformApply(tree: Apply)(implicit ctx: Context, info: TransformerInfo): Tree = {
-    val args1 = tree.args.map {
-      case arg: Typed if isWildcardStarArg(arg) =>
-        if (tree.fun.symbol.is(JavaDefined) && arg.expr.tpe.derivesFrom(defn.SeqClass))
-          seqToArray(arg.expr)
-        else arg.expr
-      case arg => arg
```

Comment:
Bugfix: Previous elimRepeated translation was wrong in one case, but this was not detected before.

Reply:
It's complicated, as usual. It had to do with empty arguments to Java-defined repeated parameters. We got an Array[Nothing], which is of course incompatible to other Arrays.